src/Hashids.net/Hashids.cs

```diff
@@ -201,25 +200,23 @@ public bool TryDecodeSingleLong(string hash, out long id)
         /// <inheritdoc />
         public virtual int DecodeSingle(string hash)
         {
-            var numbers = GetNumbersFrom(hash);
-
-            if (numbers.Length == 0)
-                throw new NoResultException("The hash provided yielded no result.");
+            var number = GetNumberFrom(hash);
 
-            if (numbers.Length > 1)
-                throw new MultipleResultsException("The hash provided yielded more than one result.");
-
-            return (int)numbers[0];
+            return number switch
+            {
+                -1 => throw new NoResultException("The hash provided yielded no result."),
+                _ => (int)number,
+            };
         }
 
         /// <inheritdoc />
         public virtual bool TryDecodeSingle(string hash, out int id)
         {
-            var numbers = GetNumbersFrom(hash);
+            var number = GetNumberFrom(hash);
 
-            if (numbers.Length == 1)
+            if (number > 0)
```

Comment:
Should it be number >= 0 for the case when id is 0?

Reply:
Yes 👍